crates/rpc/rpc-eth-types/src/logs_utils.rs

```diff
@@ -129,9 +131,15 @@ pub fn append_matching_block_logs<P: BlockReader>(
                 all_logs.push(log);
             }
             log_index += 1;
+            // Check if we have reached the max logs per response and return the last block if so
+            if let Some(max_logs_per_response) = max_logs_per_response {
+                if log_index >= max_logs_per_response as u64 {
+                    return Ok(Some(block_num_hash.number.saturating_sub(1)));
+                }
+            }
         }
```

Comment:
this doesn't look correct to me, this compares the index against the max logs per response. this is not what we want here.

Reply:
yes, only those should be enforced